lib/jormungandr/src/Cardano/Wallet/Jormungandr.hs

```diff
@@ -156,43 +156,43 @@ serveWallet (cfg, tr) sTolerance databaseDir hostPref listen lj beforeMainLoop =
     installSignalHandlers tr
     logInfo tr "Wallet backend server starting..."
     logInfo tr $ "Node is Jörmungandr on " <> toText (networkDiscriminantVal @n)
-    withNetworkLayer tr lj $ \case
-        Right (cp, nl) -> do
-            let nPort = Port $ baseUrlPort $ _restApi cp
-            let (_, bp) = staticBlockchainParameters nl
-            let rndTl = newTransactionLayer @'Mainnet (getGenesisBlockHash bp)
-            let seqTl = newTransactionLayer @n (getGenesisBlockHash bp)
-            let poolDBPath = Pool.defaultFilePath <$> databaseDir
-            Pool.withDBLayer cfg tr poolDBPath $ \db -> do
-                spl <- stakePoolLayer tr nl db
-                rndApi <- apiLayer tr rndTl nl
-                seqApi <- apiLayer tr seqTl nl
-                startServer tr nPort bp rndApi seqApi spl
-        Left e -> handleNetworkStartupError e
+    Server.withListeningSocket hostPref listen $ \case
```

Comment:
With this, I think the listening socket will queue connections for API requests, but they will only be accepted once the api server has started.

I do not know what the default listen queue length is.

It may cause problems if Daedalus is polling our API waiting for it to accept connections, depending on how the polling is done.

Maybe a future improvement will be to send ReplyPort only once the API server is ready to accept connections

Reply:
> Maybe a future improvement will be to send ReplyPort only once the API server is ready to accept connections

Could be :thinking: I know that the Daedalus setup works a bit like a state-machine. I think that, moving forward (mainnet, and likely, the next snapshot release as well), we'll need to review this state machine together and see exactly what's doable and how to better approach this problem.

Reply:
@rvl @KtorZ Daedalus only polls `network-information` endpoint to determine if the API is responding.

Reply:
Only after we get a response from that very endpoint we start triggering other API requests...